// Mysqlimpli/lib/MySqlImpli.js
/*
Created by Vaghula krishnan on 03-01-2015

Copyrights Designosaures

For internal purpose only 

*/
/**
 * MySqlImpl - A class to maintain underlying MySql Connections.
 *            
 * It's a module to manage and maintain the underlying MySql Connections and 
 * Proving the generic methods for running MySql queries.
 */

var MySqlImpl = (function() {

	var mysql = require('mysql');
	
	var pool = null;
	if(process.env.OPENSHIFT_MYSQL_DB_URL){
		pool = mysql.createPool (process.env.OPENSHIFT_MYSQL_DB_URL + "designo");
	}else{
		pool  = mysql.createPool ({
			host     : 'localhost',
			user     : 'root',
			database : 'pura',
			bigNumberStrings : 'true',
			connectionLimit : '100'
		});
	}

	var getConnection = function(callback) {
		pool.getConnection(function(err, connection) {
			callback(err, connection);
		});
	};

	var releaseConnection = function(connection) {
		connection.release();
	};

	var execute = function(sql, params, callback) {
		if (typeof params === 'function') {
			callback = params;
			params = null;
		}
	 else {
			pool.query(sql, params, callback);
		}
	};
var execute = function(sql, callback) {
			params = null;
			pool.query(sql, params, callback);
		}
	return {
		getConnection : getConnection,
		releaseConnection : releaseConnection,
		execute : execute
	};

})();

module.exports = MySqlImpl;
